store/src/genesis.rs

```diff
@@ -1,21 +1,110 @@
-use miden_crypto::merkle::{EmptySubtreeRoots, MmrPeaks, TieredSmt};
-use miden_node_proto::block_header;
-use miden_objects::{notes::NOTE_LEAF_DEPTH, Digest};
-
-use crate::state::ACCOUNT_DB_DEPTH;
-
-/// Generates the header of the genesis block. The timestamp is currently hardcoded to be the UNIX epoch.
-pub fn genesis_header() -> block_header::BlockHeader {
-    block_header::BlockHeader {
-        prev_hash: Some(Digest::default().into()),
-        block_num: 0,
-        chain_root: Some(MmrPeaks::new(0, Vec::new()).unwrap().hash_peaks().into()),
-        account_root: Some(EmptySubtreeRoots::entry(ACCOUNT_DB_DEPTH, 0).into()),
-        nullifier_root: Some(TieredSmt::default().root().into()),
-        note_root: Some(EmptySubtreeRoots::entry(NOTE_LEAF_DEPTH, 0).into()),
-        batch_root: Some(Digest::default().into()),
-        proof_hash: Some(Digest::default().into()),
-        version: 0,
-        timestamp: 0,
+use std::path::PathBuf;
+
+use miden_crypto::{
+    merkle::{EmptySubtreeRoots, MerkleError, MmrPeaks, SimpleSmt, TieredSmt},
+    Felt,
+};
+use miden_objects::{
+    accounts::Account,
+    notes::NOTE_LEAF_DEPTH,
+    utils::serde::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable},
+    BlockHeader, Digest,
+};
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+
+use crate::config::{APP, ORG};
+
+// FIXME: This is a duplicate of the constant in `store::state`
+pub(crate) const ACCOUNT_DB_DEPTH: u8 = 64;
+
+/// Default path at which the genesis file will be written to
+pub static DEFAULT_GENESIS_FILE_PATH: Lazy<PathBuf> = Lazy::new(|| {
+    directories::ProjectDirs::from("", ORG, APP)
+        .map(|d| d.data_local_dir().join("genesis.bin"))
```

Comment:
I'd prefer `genesis.dat` instead of `genesis.bin` as `.bin` may be confused with a binary, while `.dat` is usually associated with arbitrary data files.